lib/internal/Magento/Framework/Data/Collection/AbstractDb.php

```diff
@@ -505,8 +666,13 @@ protected function _renderOrders()
     {
         if (!$this->_isOrdersRendered) {
             foreach ($this->_orders as $field => $direction) {
+                if (isset(self::SQL_RESERVED_WORDS[strtoupper($field)])) {
```

Comment:
Just wonder. Maybe we can escape it in all cases?

Reply:
We can't blindly wrap the string in ticks. If it's a qualified column (eg `main_table.column_name`) or a function call then it will break.